samanthas_telegram_bot/api_queries/api_client.py

```diff
@@ -0,0 +1,371 @@
+                update=update, personal_info_id=personal_info_id
+            ),
+            expected_status_code=httpx.codes.CREATED,
+            logger=logger,
+            success_message=cls._generate_success_message_for_person_creation(
+                context=context, personal_info_id=personal_info_id
+            ),
+            failure_message=f"Failed to create young teacher with ID {personal_info_id}",
+            failure_logging_level=LoggingLevel.CRITICAL,
+            notify_admins_mode=SendToAdminGroupMode.SUCCESS_AND_FAILURE,
+            parse_mode_for_admin_group_message=ParseMode.MARKDOWN_V2,
+        )
+        return result is not None
+
+    @staticmethod
+    def get_json(
+        url_infix: str,
+        name_for_logger: str | None = None,
+        params: dict[str, str] | None = None,
+    ) -> Any:
+        """Function for simple synchronous GET requests with logging."""
+        if not name_for_logger:
+            name_for_logger = url_infix.replace("_", " ")
+
+        logger.info(f"Getting {name_for_logger} from the backend...")
+
+        # synchronous requests are only run at application startup, so no exception handling needed
+        response = httpx.get(f"{API_URL_PREFIX}/{url_infix}/", params=params)
+
+        data = response.json()
+        logger.info(f"...received {len(data)} {name_for_logger}.")
+
+        return data
+
+    @classmethod
+    async def get_level_of_written_test(
+        cls,
+        context: CUSTOM_CONTEXT_TYPES,  # type: ignore[valid-type]
+    ) -> str | None:
+        """Sends answers to written assessment to the backend, gets level and returns it."""
+        user_data = context.user_data  # type: ignore[attr-defined]
+
+        answer_ids: tuple[int, ...] = tuple(
+            item.answer_id for item in user_data.student_assessment_answers
+        )
+        number_of_questions = len(context.chat_data.assessment.questions)  # type: ignore[attr-defined]  # noqa
+
+        logger.info(
+            f"Chat {user_data.chat_id}: {len(answer_ids)} out of "
+            f"{number_of_questions} questions were answered. Receiving level from backend..."
+        )
+
+        data = await cls._send_to_backend(
+            context=context,
+            method=HttpMethod.POST,
+            url=API_URL_ENROLLMENT_TEST_GET_LEVEL,
+            data={"answers": answer_ids, "number_of_questions": number_of_questions},
+            expected_status_code=httpx.codes.OK,
+            logger=logger,
+            success_message="Checked result of written assessment",
+            failure_message="Failed to send results of written assessment and receive level",
+            failure_logging_level=LoggingLevel.CRITICAL,
+            notify_admins_mode=SendToAdminGroupMode.FAILURE_ONLY,
+        )
+
+        try:
+            level = typing.cast(str, data["resulting_level"])  # type: ignore[index]
+        except KeyError:
+            return None
+
+        logger.info(f"{level=}")
```

Comment:
If you will log all data above, you won't need this log.

Reply:
Well I don't log all `data` and I don't want to. By default `_send_to_backend()` will not log all `data` in case of success. So I only want to see what level the backend returned. And then, if user takes a SmallTalk test, this level will be overwritten later on, so I won't know the level of written assessment later in the conversation.